File: SimpleRPG/SimpleRPG/MonsterFactory/Enemy.cs
namespace SimpleRPG.MonsterFactory
{
    using System;
    using SimpleRPG;

    public class Enemy : Entity
    {
        public Enemy(Entity enemy = null, string name = "Normal Guinea Pig", int attack = 3, int defense = 3, int life = 25, int mana = 0, int stamina = 0)
        {
            Name = name;
            Attack = attack;
            Defense = defense;
            Life = life;
            Mana = mana;
            Stamina = stamina;
        }

        protected override void Kill()
        {
            Console.WriteLine($"{Name} died.");
        }

        protected override void TakeDamage(int incomingDamage, Entity attacker = null)
        {
            base.TakeDamage(incomingDamage);
            if (attacker != null)
            {
                base.RegularAttack(attacker);
            }
        }
    }
}

File: SimpleRPG/SimpleRPGMVC/wwwroot/js/enemy.ts
var Enemy = (function(name = "Standard Guinea Pig", attack = 5, defense = 5, life = 25, mana = 100, stamina = 100) {
    var state = {
        Name: name,
        Attack: attack,
        Defense: defense,
        Life: life,
        Mana: mana,
        Stamina: stamina
    };
    return {
        setName(name) {
            state.Name = name;
        },
        getName() {
            return state.Name;
        },
        getAttack() {
            return state.Attack;
        },
        getDefense() {
            return state.Defense;
        },
        getLife() {
            return state.Life;
        },
        getMana() {
            return state.Mana;
        },
        getStamina() {
            return state.Stamina;
        }
    };
})